modules/tinymce/src/themes/silver/main/ts/ui/core/complex/SelectDatasets.ts

```diff
@@ -21,47 +14,24 @@ export interface AdvancedSelectDataset extends UiFactoryBackstageForStyleFormats
 
 export type SelectDataset = BasicSelectDataset | AdvancedSelectDataset;
 
-const process = (rawFormats: string[]): BasicSelectItem[] => Arr.map(rawFormats, (item) => {
-  let title = item, format = item;
-  // Allow text=value block formats
-  const values = item.split('=');
-  if (values.length > 1) {
-    title = values[0];
-    format = values[1];
-  }
-
-  return { title, format };
-});
-
 const buildBasicStaticDataset = (data: Array<BasicSelectItem>): BasicSelectDataset => ({
   type: 'basic',
   data
 });
 
-export enum Delimiter {
-  SemiColon,
-  Space
-}
-
-const split = (rawFormats: string, delimiter: Delimiter): string[] => {
-  if (delimiter === Delimiter.SemiColon) {
-    return rawFormats.replace(/;$/, '').split(';');
-  } else {
-    return rawFormats.split(' ');
-  }
-};
-
 const buildBasicSettingsDataset = (editor: Editor, settingName: 'block_formats' | 'font_family_formats' | 'font_size_formats', delimiter: Delimiter): BasicSelectDataset => {
   // eslint-disable-next-line @tinymce/no-direct-editor-options
   const rawFormats = editor.options.get(settingName);
-  const data = process(split(rawFormats, delimiter));
+  const data = makeBasicSelectItems(rawFormats, delimiter);
   return {
     type: 'basic',
-    data
+    data,
   };
 };
 
 export {
   buildBasicSettingsDataset,
-  buildBasicStaticDataset
+  buildBasicStaticDataset,
+  Delimiter,
+  BasicSelectItem
```

Comment:
Since the refactored code is in the core, not moved to tests, you can just change where code that imports `Delimiter ` and `BasicSelectItem ` to `utils/Select` rather than re-export it here? There doesn't look like too many of them.

Alternatively, we do have places that export methods from modules only for testing (with a comment to that effect). So you could undo the refactoring if you prefer.